TYPO3.Flow/Classes/TYPO3/Flow/Cache/Backend/SimpleFileBackend.php

```diff
@@ -257,14 +252,7 @@ public function remove($entryIdentifier)
 
         $pathAndFilename = $this->cacheDirectory . $entryIdentifier . $this->cacheEntryFileExtension;
 
-        try {
-            $lock = new Lock($pathAndFilename);
-            unlink($pathAndFilename);
-            $lock->release();
-        } catch (\Exception $exception) {
-            return false;
-        }
-        return true;
+        return @unlink($pathAndFilename);
```

Comment:
This here is the only change of locking behaviour.

The reasoning to not lock the file for unlink is, that on Windows the unlink would then fail (can't unlink an opened file). In the worst-case, the file will be unlinked on *NIX while it is being locked in a different process, in which case the outcome is a non-existing cache entry/failed write. That should not be a problem as it will resolve itself with the next request that builds the cache.

The other way around would be worse on Win, as it would lead to the cache file never being removed.